yamllint/config.py

```diff
@@ -96,7 +97,23 @@ def parse(self, raw_content):
             except Exception as e:
                 raise YamlLintConfigError('invalid config: %s' % e)
 
-        if 'ignore' in conf:
+        if 'ignore' in conf and 'ignore-from-file' in conf:
+            raise YamlLintConfigError(
+                'invalid config: '
+                'ignore and ignore-from-file keys cannot be used together'
+            )
+        elif 'ignore-from-file' in conf:
+            if not isinstance(conf['ignore-from-file'], (str, list)):
+                raise YamlLintConfigError(
+                    'invalid config: '
+                    'ignore-from-file should contain filename(s)'
+                    ', either as a list or string'
+                )
```

Comment:
This doesn't check that the subitems are strings!

So yamllint would stall with a misconfiguration like:
```yaml
extends: default
ignore-from-file: [0]
```